### PR TITLE
Recovery to behavior

### PR DESCRIPTION
**Pull request by jwallace42**

### CODE REVIEW

File: configuration/packages/configuring-behavior-server.rst [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 
 .. _Github: https://github.com/ros-planning/navigation2/tree/main/nav2_recoveries
 
-The Recovery Server implements the server for handling recovery requests and hosting a vector of plugins implementing various C++ recoveries.
-It is also possible to implement independent recovery servers for each custom recovery, but this server will allow multiple recoveries to share resources such as costmaps and TF buffers to lower incremental costs for new behaviors.
+The Behavior Server implements the server for handling recovery behavior requests and hosting a vector of plugins implementing various C++ behaviors.
+It is also possible to implement independent behavior servers for each custom behavior, but this server will allow multiple behaviors to share resources such as costmaps and TF buffers to lower incremental costs for new behaviors.
 
-Note: the wait recovery has no parameters, the duration to wait is given in the action request.
+Note: the wait recovery behavior has no parameters, the duration to wait is given in the action request.
```
Comment (SteveMacenski) [inline configuration/packages/configuring-behavior-server.rst:14]: Add a note that pre-Rolling/Humble this was the Recovery server, not behavior. Some points as to places that it was renamed to help users

File: behavior_trees/overview/nav2_specific_nodes.rst [hunk 2]
```diff
@@ -21,7 +19,7 @@ Action Nodes
 
 * FollowPath - FollowPath Action Server Client (Controller Interface)
 
-* Spin, Wait, Backup - Recoveries Action Server Client
+* Spin, Wait, Backup - Action Server Client
```
Comment (SteveMacenski) [inline behavior_trees/overview/nav2_specific_nodes.rst:22]: Add behaviors

File: concepts/index.rst [hunk 3]
```diff
@@ -163,10 +163,10 @@ The general task in Nav2 for a controller is to compute a valid control effort t
 However, many classes of controllers and local planners exist.
 It is the goal of this project that all controller algorithms can be plugins in this server for common research and industrial tasks.
 
-Recoveries
+Behaviors
 ==========
 
-Recoveries are a mainstay of fault-tolerant systems.
+Recovery behaviors are a mainstay of fault-tolerant systems.
```
Comment (SteveMacenski) [inline concepts/index.rst:169]: Also mention this server can host non-recovery behaviors, its a generalized behavior server to share expensive resources like costmaps, TF buffers, etc and allows each behavior to act independently and with its own API

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Migration page: 

I think the migration guide needs a section regarding the behaviors recovery transition and renamed plugin names for users to know how to port the changes. 
- Launch files lifecycle manager need behavior_server not recovery_server
- yaml needs behavior_server not recovery_server
- yaml needs plugin namespaces changed
- Anything else?

**Comment by jwallace42** [comment]
> It doesn't look like you fixed many of the review comments?
> 
> > Nope, that was everything!
> 
> But be specific about what exactly to change

Like the names of the files?

File: concepts/index.rst [hunk 4]
```diff
@@ -177,6 +177,8 @@ Backing up or spinning in place, if permissible, allow the robot to move from a
 Finally, in the case of a total failure, a recovery may be implemented to call an operators attention for help.
 This can be done with email, SMS, Slack, Matrix, etc.
 
+It is important to note that the behavior server can hold any behavior that need access to costmap or TF buffer not just recovery behaviors.
```
Comment (SteveMacenski) [inline concepts/index.rst:180]:
```suggestion
It is important to note that the behavior server can hold any behavior to share access to expensive resources like costmaps or TF buffers, not just recovery behaviors. Each may have their own API.
```

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**